src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -34,48 +40,90 @@ public class UpdateNuGetConfigPackageSourcesMappings : Task
         /// </summary>
         public string[] SourceBuildSources { get; set; }
 
+        [Required]
+        public string SbrpRepoSrcPath { get; set; }
+
+        private const string SbrpCacheSourceName = "source-build-reference-package-cache";
+
         public override bool Execute()
         {
             string xml = File.ReadAllText(NuGetConfigFile);
             string newLineChars = FileUtilities.DetectNewLineChars(xml);
             XDocument document = XDocument.Parse(xml);
-            XElement pkgSrcMappingElement = document.Root.Descendants().FirstOrDefault(e => e.Name == "packageSourceMapping");
+            XElement pkgSourcesElement = document.Root.Descendants().FirstOrDefault(e => e.Name == "packageSources");
+            if (pkgSourcesElement == null)
+            {
+                Log.LogMessage(MessageImportance.Low, "Package sources are missing.");
+
+                return true;
+            }
 
+            XElement pkgSrcMappingElement = document.Root.Descendants().FirstOrDefault(e => e.Name == "packageSourceMapping");
             if (pkgSrcMappingElement == null)
             {
-                return true;
+                pkgSrcMappingElement = new XElement("packageSourceMapping");
+                document.Root.Add(pkgSrcMappingElement);
             }
 
-            // Union all package sources to get the distinct list.  These will get added to the source-build sources.
-            string[] packagePatterns = pkgSrcMappingElement.Descendants()
-                .Where(e => e.Name == "packageSource")
-                .SelectMany(e => e.Descendants().Where(e => e.Name == "package"))
-                .Select(e => e.Attribute("pattern").Value)
-                .Distinct()
-                .ToArray();
+            var allSourcesPackages = new Dictionary<string, List<string>>();
```

Comment:
Would it make sense to promote these dictionaries to instance members since they are passed around frequently?

Reply:
Yeah, I thought the same yesterday when I did a second round of refactoring.

Reply:
Fixed with https://github.com/dotnet/installer/pull/19114/commits/3f458e5268b5925b8fae42f35138b3f344b24f28